optimade/validator/validator.py

```diff
@@ -418,102 +213,462 @@ def main(self):
+            )
+
+        self.provider_prefix = (
+            base_info.dict()["meta"].get("provider", {}).get("prefix")
+        )
 
-        for endp in self.test_entry_endpoints:
+        # Set the response class for all `/info/entry` endpoints
+        self.available_json_endpoints, _ = self.get_available_endpoints(base_info)
+        for endp in self.available_json_endpoints:
+            self.response_classes[f"{info_endp}/{endp}"] = EntryInfoResponse
+
+        # Run some tests on the versions endpoint
+        self._log.debug("Testing versions endpoint %s", CONF.versions_endpoint)
+        self.test_versions_endpoint()
+        self.test_bad_version_returns_553()
+
+        # Test that entry info endpoints deserialize correctly
+        for endp in self.available_json_endpoints:
+            entry_info_endpoint = f"{info_endp}/{endp}"
+            self._log.debug("Testing expected info endpoint %s", entry_info_endpoint)
+            entry_info = self.test_info_or_links_endpoints(entry_info_endpoint)
+            if entry_info:
+                self._entry_info_by_type[endp] = entry_info.dict()
+
+        # Use the _entry_info_by_type to construct filters on the relevant endpoints
+        for endp in self.available_json_endpoints:
+            self._log.debug("Testing queries on JSON entry endpoint of %s", endp)
+            self.recurse_through_endpoint(endp)
+
+        # Test that the results from multi-entry-endpoints obey e.g. page limits
+        # and that all entries can be deserialized with the patched models
+        # These methods also set the test_ids for each type of entry, which are validated
+        # in the next loop.
+        for endp in self.available_json_endpoints:
             self._log.debug("Testing multiple entry endpoint of %s", endp)
             self.test_multi_entry_endpoint(f"{endp}?page_limit={self.page_limit}")
 
-        for endp in self.test_entry_endpoints:
+        # Test that the single IDs scraped earlier work with the single entry endpoint
+        for endp in self.available_json_endpoints:
             self._log.debug("Testing single entry request of type %s", endp)
             self.test_single_entry_endpoint(endp)
 
-        for endp in self.endpoint_mandatory_queries:
-            # skip empty endpoint query lists
-            if self.endpoint_mandatory_queries[endp]:
-                self._log.debug("Testing mandatory query syntax on endpoint %s", endp)
-                self.test_query_syntax(endp, self.endpoint_mandatory_queries[endp])
-
-        self._log.debug("Testing %s endpoint", LINKS_ENDPOINT)
-        self.test_info_or_links_endpoints(LINKS_ENDPOINT)
+        # Test that the links endpoint can be serialized correctly
+        self._log.debug("Testing %s endpoint", CONF.links_endpoint)
+        self.test_info_or_links_endpoints(CONF.links_endpoint)
 
         self.valid = not (bool(self.failure_count) or bool(self.internal_failure_count))
```

Comment:
Is there a reason to cast to `bool` here? Wouldn't the prepended `not` be enough?